qiskit_optimization/applications/sk_model.py

```diff
@@ -0,0 +1,110 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""An application class for the Sherrington Kirkpatrick (SK) model."""
+
+from typing import Union, List, Optional
+
+import networkx as nx
+import numpy as np
+from docplex.mp.model import Model
+
+from .graph_optimization_application import OptimizationApplication
+from ..algorithms import OptimizationResult
+from ..problems.quadratic_program import QuadraticProgram
+from ..translators import from_docplex_mp
+
+
+class SKModel(OptimizationApplication):
+    r"""Optimization application of the "Sherrington Kirkpatrick (SK) model" [1].
+
+    The SK Hamiltonian over n spins is given as:
+    :math:`H(x)=-1/\sqrt{n} \sum_{i<j} w_{i,j}x_ix_j`,
+    where :math:`x_i\in\{\pm 1\}` and :math:`w_{i,j}\in\{\pm 1\}` are chosen independently and uniformly
+    at random. Notice that there are other variants e.g., with :math:`w_{i,j}` chosen from the normal
+    distribution with mean 0 and variance 1.
+
+    References:
+        [1]: Dmitry Panchenko. "The Sherrington-Kirkpatrick model: an overview",
+        https://arxiv.org/abs/1211.1094
+    """
+
+    def __init__(
+        self, num_of_sites: int, rng_or_seed: Optional[Union[np.random.Generator, int]] = None
+    ):
+        """
+        Constructor for the SK model.
+
+        Args:
+            num_of_sites: number of sites
+            rng_or_seed: numpy pseudo-random number generator or seed for default_rng(.) or None.
+            For repeatable experiments use e.g. np.random.default_rng(<SEED>).
+            None results in usage of np.random.default_rng()
```

Comment:
```suggestion
            rng_or_seed: numpy pseudo-random number generator or seed for default_rng(.) or None.
            For repeatable experiments use e.g. np.random.default_rng(<SEED>).
            None results in usage of np.random.default_rng()
```
If you look at the html generated for this it comes out incorrectly due to lack of indenting I think of the text that is part of the parameter.

![image](https://user-images.githubusercontent.com/40241007/129624097-a832d9b1-84a7-45ff-a971-436c971054da.png)

Can we say something more like the following, that maybe is a little clearer - at least that was my intent. I think the above might be confusing as to whether the user needs to pass np.random.default_rng() with SEED to get reproducible behavior. Well I guess they could since we allow a Generator to be passed but its easier to just pass the seed and the class here builds that out internally.

An optional seed for its default generator, or a NumPy random `Generator` instance to source pseudo-random numbers. By default an np.random.default_rng() generator instance is used internally where this will be created with the seed, if it was passed, to allow experiments to be reproducible.

Reply:
@woodsp-ibm thank you. I changed the text. Sorry I didn't updated it after I added the seed as an option.

Reply:
A tiny thing - there does seem to be an extra . character in there in the default_rng method call

> NumPy pseudo-random number generator or seed for np.random.default_rng(.) or None. None results in usage of np.random.default_rng().

i.e. in this this as below
```
np.random.default_rng(.) 
```

Reply:
Changed to <seed>. BTW, it is a TCS way of denoting that the function has an argument that you use.